pkg/auth/user/util_test.go

```diff
@@ -0,0 +1,108 @@
+package user
+
+import (
+	"testing"
+	"time"
+
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+func TestExtractUserLogFields_MainFieldsTransformed(t *testing.T) {
+	user := &v1.AuthStatus{
+		Id: &v1.AuthStatus_UserId{
+			UserId: "UserID",
+		},
+		AuthProvider: &storage.AuthProvider{
+			Id:   "authProviderId",
+			Name: "authProviderName",
+			Type: "authProviderType",
+		},
+		Expires: protoconv.ConvertTimeToTimestampOrNil(time.Now()),
+		UserAttributes: ConvertAttributes(map[string][]string{
+			"a": {"b"},
+		}),
+		UserInfo: &storage.UserInfo{
+			Username:     "DO",
+			FriendlyName: "Door Opener",
+			Permissions: &storage.UserInfo_ResourceToAccess{ResourceToAccess: map[string]storage.Access{
+				"Open Magic Doors":  storage.Access_READ_WRITE_ACCESS,
+				"Close Magic Doors": storage.Access_READ_ACCESS,
+			}},
+			Roles: []*storage.UserInfo_Role{
+				{
+					Name: "Admin",
+				},
+				{
+					Name: "Analyst",
+				},
+			},
+		},
+	}
+	fields := extractUserLogFields(user)
+	assert.Len(t, fields, 9)
+	assert.True(t, fields[0].(zap.Field).Equals(zap.String("userID", user.GetUserId())))
```

Comment:
nit: you could use `.Contains` here, such that you don't have to specify the order - which we don't care about.